client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -85,13 +101,26 @@ export default class EmbedViewManager extends EventEmitter {
 		const rxLink = /(^|<p>)(https?:\/\/[^\s"]+?)(<\/p>\s*|$)/gi;
 		let match;
 		while ( ( match = rxLink.exec( content ) ) ) {
-			const isMatchingPattern = list.embeds.some( ( pattern ) => pattern.test( match[ 2 ] ) );
-			if ( isMatchingPattern ) {
-				return {
-					index: match.index + match[ 1 ].length,
-					content: match[ 2 ]
-				};
+			const url = match[ 2 ];
+
+			// Disregard URL if it's not a supported embed pattern for the site
+			const isMatchingPattern = list.embeds.some( ( pattern ) => pattern.test( url ) );
+			if ( ! isMatchingPattern ) {
+				continue;
+			}
+
+			// Disregard URL if we've not yet retrieved the embed result, or if
+			// the embed result is an error
+			const embed = EmbedsStore.get( url );
+			if ( ! embed.body ) {
+				defer( () => this.fetchEmbed( url ) );
```

Comment:
Is this defer to side-step a can not dispatch in the middle of another dispatch type error?